### ISSUE TITLE
Prefix delegation enabled but prefixes not created

### ISSUE THREAD

**Issue by kstevensonnv** [aws/amazon-vpc-cni-k8s#2411]
**What happened**:
I have launched an EKS cluster with two nodes and the managed VPC CNI add-on.
The EKS cluster is in a new VPC with nothing else running in it.
Prefix delegation is not functioning.
The Networking section of both instances shows no prefixes created on ENIs, and instead secondary IP addresses are assigned.

VPC configuration:
VPC CIDR | 10.1.0.0/19 |   |   |  
-- | -- | -- | -- | --
AWS Region | eu-west-1 |   |   |  
Addressable Hosts | 8178 |   |   |  
Spare Capacity | 2046 |   |   |  
  |   |   |   |  
Subnet Type | Subnet address | Range of addresses | Useable IPs | Hosts
Public | 10.1.0.0/22 | 10.1.0.0 - 10.1.3.255 | 10.1.0.1 - 10.1.3.254 | 1022
Public | 10.1.4.0/22 | 10.1.4.0 - 10.1.7.255 | 10.1.4.1 - 10.1.7.254 | 1022
Public  | 10.1.8.0/22 | 10.1.8.0 - 10.1.11.255 | 10.1.8.1 - 10.1.11.254 | 1022
Private | 10.1.12.0/22 | 10.1.12.0 - 10.1.15.255 | 10.1.12.1 - 10.1.15.254 | 1022
Private |10.1.16.0/22 | 10.1.16.0 - 10.1.19.255 | 10.1.16.1 - 10.1.19.254 | 1022
Private |10.1.20.0/22 | 10.1.20.0 - 10.1.23.255 | 10.1.20.1 - 10.1.23.254 | 1022
Spare | 10.1.24.0/21 | 10.1.24.0 - 10.1.31.255 | 10.1.24.1 - 10.1.31.254 | 2046

The instance type of both nodes is m7g.large which is Nitro.
The managed VPC CNI add-on has these additional configuration values:
```
{"env":{"ENABLE_PREFIX_DELEGATION":"true","WARM_PREFIX_TARGET":"1"}}
```

These appear to have been picked up:
```
❯ kubectl describe ds -n kube-system aws-node | grep -i prefix
      AWS_VPC_K8S_CNI_VETHPREFIX:             eni
      ENABLE_PREFIX_DELEGATION:               true
      WARM_PREFIX_TARGET:                     1

❯ kubectl -n kube-system get pods | grep aws
aws-node-jpp2m                        1/1     Running   1 (80m ago)   156m
aws-node-wrxsm                        1/1     Running   0             156m

❯ kubectl -n kube-system describe pod aws-node-jpp2m | grep -i prefix
      AWS_VPC_K8S_CNI_VETHPREFIX:             eni
      ENABLE_PREFIX_DELEGATION:               true
      WARM_PREFIX_TARGET:                     1

❯ kubectl -n kube-system describe pod aws-node-wrxsm | grep -i prefix
      AWS_VPC_K8S_CNI_VETHPREFIX:             eni
      ENABLE_PREFIX_DELEGATION:               true
      WARM_PREFIX_TARGET:                     1
```

Without prefix delegation enabled allocatable pods should be 29.
With prefix delegation enabled allocatable pods should be 110.

```
❯ AWS_REGION=eu-west-1 ./max-pods-calculator.sh --instance-type m7g.large --cni-version 1.12.5-eksbuild.2
29

❯ AWS_REGION=eu-west-1 ./max-pods-calculator.sh --instance-type m7g.large --cni-version 1.12.5-eksbuild.2 --cni-prefix-delegation-enabled
110
```

Both nodes have an allocatable pod limit of 110:
```
kubectl get nodes -o=jsonpath="{range .items[*]}{.status.allocatable.pods}{'\n'}{end}"
110
110
```

ipamd.log shows an IP pool of 27 and 0 prefixes.
```
{"level":"debug","ts":"2023-06-07T00:04:28.058Z","caller":"ipamd/ipamd.go:1272","msg":"IP pool stats: Total IPs/Prefixes = 27/0, AssignedIPs/CooldownIPs: 6/0, c.maxIPsPerENI = 9"}
```

Scaling nodes to 0 and launching new nodes exhibits the same issue.

[ipamd.log a test Kubernetes deployment is scaled to 10 pods](https://github.com/aws/amazon-vpc-cni-k8s/files/11671465/ipamd_k8s_deployment_10.log)
[ipamd.log a test Kubernetes deployment is scaled to 20 pods](https://github.com/aws/amazon-vpc-cni-k8s/files/11671467/ipamd_k8s_deployment_20.log)
[iipamd.log a test Kubernetes deployment is scaled to 50 pods](https://github.com/aws/amazon-vpc-cni-k8s/files/11671468/ipamd_k8s_deployment_50.log)
[pod events](https://github.com/aws/amazon-vpc-cni-k8s/files/11671543/pod_events.log)

The subnets are not exhausted of addresses.
986, 1015, and 985 addresses are free.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.26 (v1.26.4-eks-0a21954)
- CNI Version: v1.12.5-eksbuild.2
- OS (e.g: `cat /etc/os-release`): Bottlerocket OS 1.14.1 (aws-k8s-1.26)
- Kernel (e.g. `uname -a`): Linux ip-10-1-1-246.eu-west-1.compute.internal 5.15.108 1 SMP Wed May 24 23:53:57 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux

**Comment by jdn5126** [comment]
Hi @kstevensonnv , at the time that `v1.12.5` shipped, it looks like `m7g.large` was new and not properly denoted as a Nitro instance: https://github.com/aws/amazon-vpc-cni-k8s/blob/v1.12.5/pkg/awsutils/vpc_ip_resource_limit.go#L426

This was fixed in `v1.12.6`: https://github.com/aws/amazon-vpc-cni-k8s/blob/v1.12.6/pkg/awsutils/vpc_ip_resource_limit.go#L427

As for how this happened, it looks like we were not thorough enough in reviewing https://github.com/aws/amazon-vpc-cni-k8s/pull/2250 , so sorry about that 😢 . Can you confirm that this works when you upgrade to `v1.12.6-eksbuild.2`?

**Comment by kstevensonnv** [comment]
Hi @jdn5126, I have updated to v1.12.6 and prefixes are now being created as expected.

Thank you :)
